src/udev/udevd.c

```diff
@@ -1272,8 +1291,12 @@ static int on_post(sd_event_source *s, void *userdata) {
 
         if (!hashmap_isempty(manager->workers)) {
                 /* there are idle workers */
-                log_debug("Cleanup idle workers");
-                manager_kill_workers(manager);
+                if (!manager->kill_workers_event) {
+                        r = sd_event_add_time(manager->event, &manager->kill_workers_event, CLOCK_MONOTONIC,
+                                              now(CLOCK_MONOTONIC) + 3 * USEC_PER_SEC, USEC_PER_SEC, on_kill_workers_event, manager);
+                        if (r < 0)
+                                log_warning_errno(r, "Failed to add timer event: %m");
+                }
```

Comment:
i wonder if it wouldn't be nicer to make this a a timer event that just stays around after the first time it is allocated, and we just enable/disable it whenever we need.

i.e. a logic of allocating it when used first, and then enabling it with SD_EVENT_ONESHOT each time we need it would be quite nice, and reduce the number of needed allocations a bit.